src/NSFW.cpp

```diff
@@ -291,16 +291,18 @@ NSFW::StopWorker::StopWorker(NSFW *nsfw, Callback *callback):
 
 void NSFW::StopWorker::Execute() {
   uv_mutex_lock(&mNSFW->mInterfaceLock);
-
   if (mNSFW->mInterface == NULL) {
     uv_mutex_unlock(&mNSFW->mInterfaceLock);
     return;
   }
+  uv_mutex_unlock(&mNSFW->mInterfaceLock);
```

Comment:
the lock manage here is really difficult. I think we can do better, but for another PR. Just to talk about what I'm currently mulling over...

We should consider building a mapping interface between the native watcher and the javascript head. When an NSFW is created on the js side, it should assign itself the next id and create a native file watcher. The native file watcher should know the id or have a callback which knows of the id and be able to reference the js side through that id (when performing callbacks). When the nsfw module itself halts, we should let the native side clean up on its own time, but sever the map between the native and js layer... That way we don't need to sync at all in the stop worker!

What do you think? Just thoughts for another PR I think.